nix/nixos/smash-service.nix

```diff
@@ -69,22 +90,32 @@ in {
     };
   };
   config = lib.mkIf cfg.enable {
-    services.smash.script = pkgs.writeShellScript "smash" ''
+    services.smash.script = let
+    in pkgs.writeShellScript "smash" ''
       set -euo pipefail
 
-      cp ${cfg.postgres.pgpass} $RUNTIME_DIRECTORY/pgpass
+      RUNTIME_DIRECTORY=''${RUNTIME_DIRECTORY:-$(pwd)}
+      ${if (cfg.socketPath == null) then ''if [ -z ''${CARDANO_NODE_SOCKET_PATH:-} ]
+      then
+        echo "You must set \$CARDANO_NODE_SOCKET_PATH"
+        exit 1
+      fi'' else "export CARDANO_NODE_SOCKET_PATH=\"${cfg.socketPath}\""}
+
+      ${lib.optionalString cfg.postgres.generatePGPASS ''
+      cp ${cfg.postgres.pgpass} /$RUNTIME_DIRECTORY/pgpass
       chmod 0600 $RUNTIME_DIRECTORY/pgpass
-      export SMASHPGPASSFILE=$RUNTIME_DIRECTORY/pgpass
+      export SMASHPGPASSFILE=/$RUNTIME_DIRECTORY/pgpass
+      ''}
 
       ${cfg.package}/bin/smash-exe run-migrations --mdir ${../../schema}
       exec ${cfg.package}/bin/smash-exe run-app-with-db-sync \
         --config ${configFile} \
-        --socket-path ${cfg.socketPath} \
+        --socket-path "$CARDANO_NODE_SOCKET_PATH" \
         --schema-dir ${../../schema}
     '';
     environment.systemPackages = [ cfg.package config.services.postgresql.package ];
     systemd.services.smash = {
-      path = [ cfg.package config.services.postgresql.package pkgs.netcat ];
+      path = [ cfg.package pkgs.netcat pkgs.postgresql ];
```

Comment:
just use postgresql from pkgs instead of the version used by service. This may cause issues, but again, hard to work-around when not used with postgresql nixos service (like with script).